EIPS/eip-6963.md

```diff
@@ -234,11 +235,21 @@ function onPageLoad(): EIP6963ProviderDetail[] => {
 
 ## Security Considerations
 
-The security considerations of [EIP-1193](./eip-1193.md) apply to this EIP.
+### EIP-1193 Security considerations
+
+The security considerations of [EIP-1193](./eip-1193.md) apply to this EIP. Implementers are expected to consider and follow the guidance of the providers they're utilizing as well.
+
+### Prototype Pollution of Wallet Provider objects
+
+Browser extensions, and therefore Wallet extensions, are able to modify the contents of the page and the Provider object by design. Given the provider objects of various Wallets are considered high trusted interface to communicate transaction data. In order to prevent the page or various other extensions from modifying the interaction between the DApp and the Wallet in an unexpected way. The expected way to handle this is to utilize `object.freeze()` on the  `EIP1193Provider` object before the wallet dispatches the `eip6963:announceProvider` Event.
+
+### Prevent SVG Javascript Execution 
+
+The use of SVG images introduces a cross-site scripting risk as they can include JavaScript code. This Javascript executes within the context of the page and can therefore modify the page or the contents of the page. So when considering the experience of rendering the icons, DApps need to take into consideration how they'll approach handling these concerns in order to prevent an image being used as an obfuscation technique to hide malicious modifications to the page or to other wallets.
 
-The use of SVG images introduces a cross-site scripting risk as they can include JavaScript code. Applications and libraries must render SVG images using the `<img>` tag to ensure no JS executions can happen.
+### Prevent Wallet Fingerprinting
 
-// TODO for @kdenhartog: Add a privacy section about Wallet fingerprinting and how Wallets should not announce themselves until the DApp has dispatched the request provider Event and gathered consent from the user that they wish to announce themselves to the DApp.
+One advantage to the concurrency Event loop utilized by this design is that it operates in a manner where either the DApp or the Wallet can initiate the flow to announce a provider. For this reason, Wallet implementers can now consider whether or not they wish to announce themselves to all pages or attempt alternative means in order to reduce the ability for a user to be fingerprinted by the injection of the `window.ethereum` object. Some examples, of alternative flows to consider would be to wait to inject the provider object until the DApp has announced the `eip6963:requestProvider`. At that point, the wallet can initiate a UI consent flow to ask the user if they would like to share their wallet address. This allows for the Wallet to enable the option of a "private connect" feature. However, if this approach is taken Wallet's must also consider how they intend to support backwards compatability with a DApp that does not support this EIP.
```

Comment:
This is an awesome side-feature, but I'm still concerned as to how we educate dapps to make sure they react to the announcements that are emitted later than on the initial page load phase.

We can imagine a wallet exposing itself not after the `eip6963:requestProvider` event, but after a deliberate user action provided by wallet UI, e.g. a dialog confirming with the user whether they want to announce themselves to the current website.

Reply:
> This is an awesome side-feature, but I'm still concerned as to how we educate dapps to make sure they react to the announcements that are emitted later than on the initial page load phase. 

With the new concurrent model, DApps will automatically behave in this way by registering their event handler on page load and then firing the `eip6963:requestProvider` event afterwards. As long as that event handler is registered by the DApp then it is able to respond accordingly. This means that in some cases DApp's may still prompt the user to click the `connect` button after already providing consent and intiating the flow in the wallet, which will feel a bit clunky but it will work. Alternatively, the DApp's could optimistically fire the `requestAccounts` call right as soon as a provider is chosen in which case because the user already granted the consent it would skip prompting the user and return the wallet address.

This would also work in the same way with the legacy flow. For example, under the current presumed flow here's how it works:

1. User clicks connect button which calls `async window.ethereum.request({ method: 'eth_requestAccounts' })`
2. User consents to the account being provided
3. Wallet supplies the address back in a promise structure during success or promise reject when no consent is given

Under the new model, here's how it would work:

1. User clicks "inject" button (not the best name for UI) in wallet to inject it and selects the address
2. User clicks "connect" button on the DApp
3. DApp responds with the wallet address in a promise immediately because the User has already consented

In the case where step 1 is skipped the wallet would not connect in this new "Private" mode. That's what makes this an implementation considerations is because both flows can be executed and supported using the same interface, but the implementation logic differs, defaults can differ, and configuration logic can differ but we can support the new and the old flow using this pattern AND potentially address fingerprinting concerns if implemented this way.

Does that make sense?

Reply:
The event system makes perfect sense, I'm just beware of cases where dapps "bail out" of listening after hearing first wallet announcement or once hearing a "desired" wallet announced.

Reply:
Ahh, that's a good point we should probably add something to the text to state that must not be done.

Reply:
addressed in latest changes, so will be resolving this one.